test/pp_api_test/test_imgcodec.cc

```diff
@@ -78,4 +116,55 @@ TEST(ImageDecoderTest, TestJpegDecoder) {
   out_range = out_tensor.Data() + 875 * 1300 * 3 + 1296 * 3;
   ASSERT_EQ(std::vector<uint8_t>(out_range, out_range + 12),
             std::vector<uint8_t>({208, 210, 197, 204, 206, 193, 198, 200, 187, 194, 196, 183}));
+  #endif
+#else
+  out_range = out_tensor.Data() + 1296 * 3;
+  ASSERT_EQ(std::vector<uint8_t>(out_range, out_range + 12),
+            std::vector<uint8_t>({221, 237, 224, 225, 236, 219, 218, 222, 199, 203, 202, 174}));
+
+  out_range = out_tensor.Data() + 438 * 1300 * 3;
+  ASSERT_EQ(std::vector<uint8_t>(out_range, out_range + 12),
+            std::vector<uint8_t>({84, 68, 55, 86, 70, 55, 92, 77, 58, 101, 86, 65}));
+
+  out_range = out_tensor.Data() + 875 * 1300 * 3 + 1296 * 3;
+  ASSERT_EQ(std::vector<uint8_t>(out_range, out_range + 12),
+            std::vector<uint8_t>({208, 210, 197, 204, 206, 193, 198, 200, 187, 194, 196, 183}));
+#endif
+}
+
+#if OCOS_ENABLE_VENDOR_IMAGE_CODECS
+TEST(ImageDecoderTest, TestTiffDecoder) {
```

Comment:
This is just an example. Many other formats can be supported as long as the OS provides such capability.